src/named_source.rs

```diff
@@ -3,6 +3,7 @@ use crate::{MietteError, MietteSpanContents, SourceCode, SpanContents};
 /// Utility struct for when you have a regular [`SourceCode`] type that doesn't
 /// implement `name`. For example [`String`]. Or if you want to override the
 /// `name` returned by the `SourceCode`.
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct NamedSource<S: SourceCode + 'static> {
```

Comment:
I'm confused: doesn't this mean that `S` is now required to implement _all_ of those? I'm confused as to why they don't have to show up in the constraints. This is, of course, a silly Rust question.

Reply:
Hopefully it's not too silly of a question — I had the same! It looks like the derive macros do the clever thing though and add a trait-bound on the generic — it means that if `S` implements `Clone`, then so will `NamedSource`, but if it doesn't, that's no big deal and `NamedSource` won't either.

`cargo-expand` shows: 
![image](https://github.com/zkat/miette/assets/6251883/c13b6ade-0ed7-4d48-9b72-d812534209b9)

Reply:
that's incredible!